src/communication/xcSession.ts

```diff
@@ -20,7 +20,8 @@ export interface Session {
     setPrivateTopic(privateTopic: string): void;
     addPrivateTopic(privateTopic: string): void;
     removePrivateTopic(privateTopic: string): void;
-    init(openListener: (e: Event) => void, errorListener: (err: Error) => void, closeListener: (e: CloseEvent) => void): void;
+    init(): Promise<Event>;
+    getUnexpectedCloseSessionError(): Promise<Error>;
```

Comment:
Use a callback as parameter instead of a promise
Improve the naming

Reply:
i don't get the point of the init method... what does it do?
isn't the session 'initted' when we get it?

Reply:
done